plansys2_problem_expert/src/plansys2_problem_expert/Utils.cpp

```diff
@@ -184,11 +184,11 @@ std::tuple<bool, bool, double> evaluate(
             }
             break;
           case parser::pddl::tree::ARITH_MULT:
-            return std::make_tuple(true, true, std::get<2>(left) * std::get<2>(right));
+            return std::make_tuple(true, false, std::get<2>(left) * std::get<2>(right));
```

Comment:
this was changed to make it match the boolean results from the FUNCTION_MODIFIER section (which are all arithmetic operations, and they all return false for the boolean result. By my understanding the values in the tuple represent the following: <operation successful, result of any boolean operation (like >=), result of any arithmetic operation>

Reply:
Ok, then, if we want to match the behavior of "C" like language, the boolean `true` correspond to "non-zero".
In this case it can have no meaning since the I thing that the usage of an arithmetic expression instead of a boolean one should lead to a syntax error in PDDL. So the implementation you propose still correct, but have no effect. Am I correct ?

Reply:
Hi @teyssieuman. I thought I'd weigh in here since I wrote that section of code. Yes, I believe the implementation is correct. For arithmetic operations, the second element of the tuple has no effect. Likewise, for boolean operations, the third element of the tuple has no effect. Thus for arithmetic operations we set the value of the second tuple element to be false (a value that really doesn't mean anything, but seems slightly better than setting it to true). Likewise for boolean operations we set the third element to 0 (a value that again doesn't really mean anything). Note that a description of the evaluate function is provided in the header, which hopefully explains all of this. Thanks for reviewing this code! We're happy to respond to any more questions.

https://github.com/IntelligentRoboticsLabs/ros2_planning_system/blob/a495a6825dc18e8cffdbf6f5ea2fc638468cbe41/plansys2_problem_expert/include/plansys2_problem_expert/Utils.hpp#L32

Reply:
Thank you @jjzapf for this clear answer ...